File: CheckDigitOrNot.java
import java.util.*;


class CheckDigitOrNotX
{
    public char Ch;

    public CheckDigitOrNotX(char CValue)
    {
        Ch = CValue;
    }

    public boolean CheckDigit()
    {
        if((Ch >= '0') && (Ch <= '9'))
        {
            return true;
        }
        else
        {
            return false;
        }
    }
}

class CheckDigitOrNot
{
    public static void main(String arg[])
    {
        char cValue;
        boolean bRet = false;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the character:");
        cValue = sobj.nextLine().toCharArray()[0];

        CheckDigitOrNotX cobj = new CheckDigitOrNotX(cValue);
        bRet = cobj.CheckDigit();
        if(bRet == true)
        {
            System.out.println(cValue + " is a digit.");
        }
        else
        {
            System.out.println(cValue + " is not digit.");
        }
    }
}

File: CheckDigitOrNot.c
#include<stdio.h>
#include<stdbool.h>

bool CheckDigit(char ch)
{
    if((ch >= '0')&&(ch <= '9'))
    {
        return true;
    }
    else
    {
        return false;
    }
}

int main()
{
    char cValue = '\0';
    bool bRet = false;

    printf("Enter one digit:\n");
    scanf("%c",&cValue);

    bRet = CheckDigit(cValue);
    if(bRet == true)
    {
        printf("%c number is digit.\n",cValue);
    }
    else
    {
        printf("%c number is not digit.\n",cValue);
    }

    return 0;
}